code/datums/character_preview.dm

```diff
@@ -0,0 +1,119 @@
+datum/character_preview
+	var/global/max_preview_id = 0
+	var/preview_id
+	var/window_id
+	var/client/viewer
+	var/obj/screen/handler
+	var/mob/living/carbon/human/preview_mob
+
+	window
+		New(client/viewer)
+			var/winid = "preview_[max_preview_id]"
+
+			winclone(viewer, "blank-map", winid)
+
+			winset(viewer, winid, list2params(list(
+				"size" = "128,128",
+				"title" = "Character Preview",
+				"can-close" = FALSE,
+				"can-resize" = FALSE,
+			)))
+
+			. = ..(viewer, winid)
+
+		disposing()
+			. = ..()
+			SPAWN_DBG(0)
+				if (src.viewer)
+					winset(src.viewer, "[src.window_id]", "parent=")
+
+		proc/Show(shown = TRUE)
+			winshow(src.viewer, src.window_id, shown)
+
+	multiclient
+		var/list/viewers = list()
+
+		New(control_id = null)
+			. = ..(null, "unused", control_id)
+
+		disposing()
+			for (var/client/viewer in src.viewers)
+				if (viewer)
+					viewer.screen -= src.handler
+					viewer.screen -= src.preview_mob
+			. = ..()
+
+		proc/AddClient(client/viewer)
+			if (viewer && !(viewer in src.viewers))
+				src.viewers += viewer
+				viewer.screen += src.handler
+				viewer.screen += src.preview_mob
+
+		proc/RemoveClient(client/viewer)
+			if (viewer && (viewer in src.viewers))
+				src.viewers -= viewer
+				viewer.screen -= src.handler
+				viewer.screen -= src.preview_mob
+
+		proc/RemoveAllClients()
+			for (var/client/viewer in src.viewers)
+				if (viewer)
+					viewer.screen -= src.handler
+					viewer.screen -= src.preview_mob
+			src.viewers.len = 0
+
+	New(client/viewer, window_id, control_id = null)
```

Comment:
this is really gross-ly relatively pathed, with this being below all of the children and their News